### DISCUSSION TITLE
对wp功能影响尽量最小的情况下的Cloudflare页面规则

### DISCUSSION THREAD

**Discussion by murasame-zako** [mirai-mamori/Sakurairo#851] [Show and tell]
## 介绍

为了合理利用Cloudflare CDN的缓存与安全功能，需要正确配置页面规则。免费用户只有三条页面规则，但是这根本不够用，需要对性能或者安全性做出取舍。经过测试，我分享两种页面规则。

### 节约一条页面规则

请先安装 [WPS隐藏登陆](https://cn.wordpress.org/plugins/wps-hide-login/) 插件，将登陆页面的/wp-login.php移动到例如“/wp-admin-123456”下。可以让登陆页面也得到安全保护，防止暴力尝试对服务器造成负担。与/wp-admin共同接受同一条页面规则的保护。

### 第一种页面规则

1. 域名/wp-admin/admin-ajax.php
安全级别: 低, 缓存级别: 绕过, 电子邮件混淆技术: 关
2. 域名/wp-admin*
浏览器完整性检查: 开, 安全级别: I'm Under Attack, 缓存级别: 绕过
3. 域名/*
浏览器缓存 TTL: 2 小时, 缓存级别: 缓存所有内容, 边缘缓存 TTL: 2 小时, 电子邮件混淆技术: 开, 自动 HTTPS 重写: 开, 缓存欺骗盔甲: 开, 源服务器缓存控制: 开

第一条规则是因为第二条规则开启了wp-admin的5秒盾，如果不绕过，就导致评论无法正确提交，提交评论的接口将变为5秒盾界面。

### 第二种页面规则

1. 域名/?s=*
浏览器完整性检查: 开, 安全级别: I'm Under Attack, 缓存级别: 绕过
2. 域名/wp-admin*
浏览器完整性检查: 开, 安全级别: I'm Under Attack, 缓存级别: 绕过
3. 域名/*
浏览器缓存 TTL: 2 小时, 缓存级别: 缓存所有内容, 边缘缓存 TTL: 2 小时, 电子邮件混淆技术: 开, 自动 HTTPS 重写: 开, 缓存欺骗盔甲: 开, 源服务器缓存控制: 开

这个规则无法正确提交评论，适合禁用评论的博客使用，比前一个规则提高了安全性，降低机器人爬取搜索接口导致的占用。

如果有办法能够把搜索接口调整到/wp-admin下还能节省一条页面规则，兼顾性能和安全性。

两种规则都不太完美，存在一些bug，就是wp预览没办法正确刷新。在编写评论前必须在登陆后台后再打开文章页面，否则即使登陆了也因为cloudflare的页面缓存导致只能游客身份留下评论。三条页面规则真不够用。

**Comment by CRK1918** [comment]
非常感谢你的分享。我为/*只关了Rocket Loader™ 规则，好像没问题，现在我得重新看一下评论功能了。
